### PR TITLE
Support for custom types on the client

### PR DESCRIPTION
**Pull request by einarmo**
Improved support for custom codegen on the client, with utils for reading custom types from the server.

This hit on a bug due to missing codegen which lead me into a rabbit-hole that concluded with me finding that we were running codegen on the wrong schema version.

### CODE REVIEW

File: code_gen_config.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ extra_header: |
   // Copyright (C) 2017-2024 Adam Lock, Einar Omang
 targets:
   - type: types
-    file_path: schemas/1.0.4/Opc.Ua.Types.bsd
+    file_path: schemas/1.05/Opc.Ua.Types.bsd
```
Comment (svanharmelen) [inline code_gen_config.yml:7]: Shouldn't this be 1.0.5?

**Comment by svanharmelen** [comment]
I've noticed your still driving this package all by yourself. Which in a way maybe makes sense as you have the most holistic view of the whole code base, but on the other hand introducing shared responsibility and the "4-eyes principle" for any merged PRs is probably good for the health of the project (at least in the long run).

If your open to it I'm more then willing to step up and help maintain the package. You'll be the lead, but I can help review your stuff and respond to other issues and/or PRs that don't make architectural changes (I'll leave those with you).

I'm just laying this out there, I'm in no way offended or anything like that if you prefer not to change the current setup or just don't think I'm the right person to help out! So just let me know your honest opinion 😉

File: code_gen_config.yml (hunk 1 above)
Comment (einarmo) [reply code_gen_config.yml:7]: It's actually called 1.05, the version of the standard, that is. Either way this is just the path in the repo.

**Comment by einarmo** [comment]
@svanharmelen absolutely. It would be pretty messed up of me to fork a repo for only having one maintainer then refusing other maintainers. I'll make you one, slightly unavailable at the moment but I'll get it done today or tomorrow.

**Comment by einarmo** [comment]
There we go. Invited you to the Opcua Rust team in the freeopcua org. That should make you maintainer of this repo if I didn't do something wrong.

**Comment by oroulet** [comment]
@svanharmelen I invited you. more motivated people is great

**Comment by svanharmelen** [comment]
Thanks, got the invite and joined 👍🏻 Looking forward to help build/maintain the crate...

**Pull request merged**